integration/embed_test.go

```diff
@@ -43,9 +48,9 @@ func TestEmbedEtcd(t *testing.T) {
 	}
 
 	// 4000x so no conflict with e2e tests
-	url1, _ := url.Parse("http://localhost:40000")
-	url2, _ := url.Parse("http://localhost:40001")
-	url3, _ := url.Parse("http://localhost:40002")
+	url1, _ := url.Parse(fmt.Sprintf("http://localhost:%d", embedBasePort))
```

Comment:
Still needs new URLs for each `setupEmbedCfg`, otherwise the will be port reuse among the subtests. This part should probably be something like `func newEmbedURLs() []url.URL { ... }` which is called for each `setupEmbedCfg`.